### PR TITLE
Policy Packs for AWS .

### PR DESCRIPTION
**Pull request by niharika2k00**
image

image

image

image

image

### CODE REVIEW

File: hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/policies.tf [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  template_input = <<-EOT
+    {
+      volume {
+        attachments: get(path: "Attachments")
+        encrypted: get(path: "Encrypted")
+      }
+    }
+    EOT
+  template       = <<-EOT
+    {%- set attachedWithInstance = false -%}
```
Comment (SumitPopat) [inline hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/policies.tf:43]: Please indent these correctly. [Example](https://github.com/turbot/guardrails-samples/blob/main/cis_enforcements/aws/cis_v300_s1_iam/login_profile.tf#L37).

File: hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/policies.tf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        volume {
+          attachments: get(path: "Attachments")
+          encrypted: get(path: "Encrypted")
+        }
+      }
+      EOT
+  template       = <<-EOT
+    {%- set attachedWithInstance = false -%}
+    {%- for attachment in $.volume.attachments -%}
+    {#- If AWS EBS volume is attached with EC2 instance -#}
```
Comment (SumitPopat) [inline hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/policies.tf:16]:
```suggestion
    {#- If AWS EBS volume is attached to an EC2 instance -#}
```

Comment (SumitPopat) [inline hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/policies.tf:16]: Please indent these correctly. [Example](https://github.com/turbot/guardrails-samples/blob/main/cis_enforcements/aws/cis_v300_s1_iam/login_profile.tf#L37).

File: hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/README.md
Comment (SumitPopat) [inline hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/README.md:1]: I think the Policy Pack name should be `detect_encryption_at_rest_enabled_for_attached_ebs_volume`

File: hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/common.tf [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Policy Pack
+resource "turbot_smart_folder" "pack" {
+  title       = "Detect whether encryption is enabled for AWS EBS volumes attached to AWS EC2 instances"
+  description = "Detect if encryption is enabled for AWS EBS volumes attached to EC2 instances"
```
Comment (SumitPopat) [inline hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/common.tf:4]:
```suggestion
  title       = "Detect if encryption at rest is enabled for AWS EBS volumes attached to AWS EC2 instances"
  description = "Detect if encryption at rest is enabled for AWS EBS volumes attached to AWS EC2 instances"
```

File: hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/README.md
Comment (niharika2k00) [reply hub_content/policy_packs/aws/ec2/detect_encryption_enabled_for_attached_ebs_volumes/README.md:1]: ok

File: hub_content/policy_packs/azure/network/detect_nsg_allow_all_or_rdp_or_ssh_inbound_access/policies.tf [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Azure > Network > Network Security Group > Approved
+resource "turbot_policy_setting" "azure_network_security_group_ingress_rules_approved" {
+  resource = turbot_smart_folder.pack.id
+  type     = "tmod:@turbot/azure-network#/policy/types/networkSecurityGroupIngressRulesApproved"
+  value    = "Check: Approved"
+  # value    = "Enforce: Delete unapproved"
```
Comment (rajlearner17) [inline hub_content/policy_packs/azure/network/detect_nsg_allow_all_or_rdp_or_ssh_inbound_access/policies.tf:6]: @niharika2k00 is this right? or should be Enforce: Delete unapproved if new

```
Skip
Check: Approved
Enforce: Delete unapproved if new
```

File: hub_content/policy_packs/azure/network/detect_nsg_allow_all_or_rdp_or_ssh_inbound_access/policies.tf [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+# Azure > Network > Network Security Group > Custom
+resource "turbot_policy_setting" "azure_network_security_group_ingress_rules_approved_rules" {
+  resource = turbot_smart_folder.pack.id
+  type     = "tmod:@turbot/azure-network#/policy/types/networkSecurityGroupIngressRulesApprovedRules"
+  template = <<-EOT
+    APPROVE $.turbot.fromPort:=3389 $.turbot.toPort:=3389
+    APPROVE $.turbot.fromPort:=22 $.turbot.toPort:=22
+    APPROVE *
+    EOT
```
[the remark this answers is not in the document]
Comment (niharika2k00) [reply hub_content/policy_packs/azure/network/detect_nsg_allow_all_or_rdp_or_ssh_inbound_access/policies.tf:17]: yes updated. In PR : https://github.com/turbot/guardrails-samples/pull/732

File: hub_content/policy_packs/aws/ec2/detect_imdsv2_enforced_for_instance/policies.tf [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# AWS > EC2 > Instance > Approved
+resource "turbot_policy_setting" "aws_ec2_instance_approved" {
```
[the remark this answers is not in the document]
Comment (niharika2k00) [reply hub_content/policy_packs/aws/ec2/detect_imdsv2_enforced_for_instance/policies.tf]: updated

**Review by SumitPopat — APPROVED** [review]

**Pull request merged**